src/python/pants/invalidation/cache_manager.py

```diff
@@ -117,6 +120,21 @@ def previous_results_dir(self):
       raise ValueError('There is no previous_results_dir for: {}'.format(self))
     return self._previous_results_dir
 
+  def ensure_legal(self):
+    """Return True as long as the state does not break any internal contracts."""
+    # Do our best to provide complete feedback, it's easy to imagine the frustration of flipping between error states.
+    if self._results_dir:
+      errors = ''
+      if not os.path.islink(self._results_dir):
+        errors += '\nThe results_dir is no longer a symlink:\n\t* {}'.format(self._results_dir)
+      if not os.path.isdir(self._current_results_dir):
+        errors += '\nThe current_results_dir directory was not found\n\t* {}'.format(self._current_results_dir)
+      if errors:
+        raise self.IllegalResultsDir(
+          '\nThe results_dirs state should not be manually cleaned or recreated by tasks.\n{}'.format(errors)
+        )
+    return True
```

Comment:
Yes...I added the return value because things like this often end up being useful as bools, and people naturally would try it.  Like added to a `should_cache_artifacts` method or something. It was an attempt to conform to expectation. Without the `return True`, `vt.ensure_legal() would be Falsey, which felt confusing.

I would prefer to leave it, but will remove if you let me know you disagree.